client/app/pages/queries-list/QueriesList.jsx

```diff
@@ -0,0 +1,205 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { react2angular } from 'react2angular';
+
+import { PageHeader } from '@/components/PageHeader';
+import { Paginator } from '@/components/Paginator';
+import { QueryTagsControl } from '@/components/tags-control/QueryTagsControl';
+import { SchedulePhrase } from '@/components/queries/SchedulePhrase';
+
+import LiveItemsList from '@/components/items-list/LiveItemsList';
+import LoadingState from '@/components/items-list/components/LoadingState';
+import * as Sidebar from '@/components/items-list/components/Sidebar';
+import ItemsTable, { Columns } from '@/components/items-list/components/ItemsTable';
+
+import { Query } from '@/services/query';
+import { currentUser } from '@/services/auth';
+import navigateTo from '@/services/navigateTo';
+import { routesToAngularRoutes } from '@/lib/utils';
+
+import QueriesListEmptyState from './QueriesListEmptyState';
+
+import './queries-list.css';
+
+class QueriesList extends React.Component {
+  static propTypes = {
+    currentPage: PropTypes.string.isRequired,
+  };
+
+  static routes = [
+    {
+      path: '/queries',
+      title: 'Queries',
+      key: 'all',
+    },
+    {
+      path: '/queries/favorites',
+      title: 'Favorite Queries',
+      key: 'favorites',
+    },
+    {
+      path: '/queries/archive',
+      title: 'Archived Queries',
+      key: 'archive',
+    },
+    {
+      path: '/queries/my',
+      title: 'My Queries',
+      key: 'my',
+    },
+  ];
+
+  static sidebarMenu = [
+    {
+      key: 'all',
+      href: 'queries',
+      title: 'All Queries',
+    },
+    {
+      key: 'favorites',
+      href: 'queries/favorites',
+      title: 'Favorites',
+      icon: () => <Sidebar.MenuIcon icon="fa fa-star" />,
+    },
+    {
+      key: 'archive',
+      href: 'queries/archive',
+      title: 'Archive',
+      icon: () => <Sidebar.MenuIcon icon="fa fa-archive" />,
+    },
+    {
+      key: 'my',
+      href: 'queries/my',
+      title: 'My Queries',
+      icon: () => <Sidebar.ProfileImage user={currentUser} />,
+      isAvailable: () => currentUser.hasPermission('create_query'),
+    },
+  ];
+
+  static listColumns = [
+    Columns.favorites({ className: 'p-r-0' }),
+    Columns.custom.sortable((text, item) => (
+      <React.Fragment>
+        <a className="table-main-title" href={'dashboard/' + item.slug}>{ item.name }</a>
+        <QueryTagsControl
+          className="d-block"
+          tags={item.tags}
+          isDraft={item.is_draft}
+          isArchived={item.is_archived}
+        />
+      </React.Fragment>
+    ), {
+      title: 'Name',
+      field: 'name',
+      width: null,
+    }),
+    Columns.avatar({ field: 'user', className: 'p-l-0 p-r-0' }, name => `Created by ${name}`),
+    Columns.dateTime.sortable({ title: 'Created At', field: 'created_at' }),
```

Comment:
If we add filters in the future, how would that look like?
Wouldn't it make sense to move it to the override obj?
```js
Column({
  dataType: Column.DATETIME,
  sortable: true,
  filterable: true,
  title: 'Created At',
  field: 'created_at'
}),
```

Reply:
So, yes - it's a good point. You already can mark column as sortable with adding property `sorter: true`, and you already can use it without calling that helper. But I decided to add it because then it's easy to see which columns are sortable. If you think that it's redundant - I'll remove it and will just use property everywhere.